distributed/tests/test_client.py

```diff
@@ -1116,6 +1117,27 @@ def g():
     result = yield y._result()
     assert result == 456
 
+@gen_cluster(client=True)
+def test_upload_file_zip(c, s, a, b):
```

Comment:
The test would be more robust if it scrubbed out `myfile` from `sys.modules` at the start and at the end.